# Perkenalan
Nama : Luthfi Fadhlillah
(FTDS Batch 009)

# Membuat Data Baru Kasus Covid-19
Instruksi: 
1. Jumlah kasus kumulativ tertinggi pada tanggal 20 Januari 2022
2. Puncak tertinggi kasus harian di Indonesia

# Import Pustaka
Melakukan import library/pustaka yang akan digunakan

In [65]:
import pandas as pd
import numpy as np

# Data Loading
Membuka data yang akan digunakan

In [66]:
data = pd.read_csv('https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_confirmed_global.csv')

In [67]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/10/22  2/11/22  \
0          0        0        0        0        0  ...   169448   169940   
1          0        0        0        0        0  ...   267020   267020   
2          0        0        0        0        0  ...   260191   260723   
3          0        0        0        0        0  ...    37074    37140   
4          0        0        0        0        0  ...    98474    98501   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      465      490   
281        0        0        0        0        0  ...    11604    11659   
282        0        0        0        0        0  ...   308556   309013   
283        0        0        0        0        0  ...   231040   231214   

     2/12/22  2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  
0     170152   170604   171246   171422   171519   171673   171857   171931  
1     267551   268008   268304   268491   268940   269301   269601   269904  
2     261226   261752   262165   262570   262994   263369   263685   263936  
3      37140    37140    37277    37361    37452    37522    37589    37589  
4      98514    98514    98514    98555    98568    98585    98605    98617  
..       ...      ...      ...      ...      ...      ...      ...      ...  
279   600128   600128   613846   618527   622175   625028   627956   627956  
280      498      501      501      505      507      507      508      508  
281    11664    11671    11679    11699    11707    11718    11731    11731  
282   309293   309502   309579   309870   310155   310474   310764   311016  
283   231299   231381   231603   231603   232213   232598   233030   233224  

[284 rows x 764 columns]

# Data Exploration
Melakukan data eksplorasi pada datasheet secara sederhana.

In [68]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/10/22  2/11/22  \
0          0        0        0        0        0  ...   169448   169940   
1          0        0        0        0        0  ...   267020   267020   
2          0        0        0        0        0  ...   260191   260723   
3          0        0        0        0        0  ...    37074    37140   
4          0        0        0        0        0  ...    98474    98501   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      465      490   
281        0        0        0        0        0  ...    11604    11659   
282        0        0        0        0        0  ...   308556   309013   
283        0        0        0        0        0  ...   231040   231214   

     2/12/22  2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  
0     170152   170604   171246   171422   171519   171673   171857   171931  
1     267551   268008   268304   268491   268940   269301   269601   269904  
2     261226   261752   262165   262570   262994   263369   263685   263936  
3      37140    37140    37277    37361    37452    37522    37589    37589  
4      98514    98514    98514    98555    98568    98585    98605    98617  
..       ...      ...      ...      ...      ...      ...      ...      ...  
279   600128   600128   613846   618527   622175   625028   627956   627956  
280      498      501      501      505      507      507      508      508  
281    11664    11671    11679    11699    11707    11718    11731    11731  
282   309293   309502   309579   309870   310155   310474   310764   311016  
283   231299   231381   231603   231603   232213   232598   233030   233224  

[284 rows x 764 columns]

In [69]:
print('Jumlah row', data.shape[0], '\nJumlah column', data.shape[1])

Jumlah row 284 
Jumlah column 764


Dapat dilihat bahwa data ini memiliki 284 baris dan 764 column

In [70]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 284 entries, 0 to 283
Columns: 764 entries, Province/State to 2/19/22
dtypes: float64(2), int64(760), object(2)
memory usage: 1.7+ MB


Data ini memiliki 2 column dengan tipe data float, 760 column dengan tipe data integer, dan 2 column dengan tipe data object

In [71]:
data.describe().T

count          mean           std       min          25%  \
Lat      282.0  2.010637e+01  2.584145e+01  -71.9499     4.643279   
Long     282.0  2.195872e+01  7.589337e+01 -178.1165   -22.036550   
1/22/20  284.0  1.961268e+00  2.640260e+01    0.0000     0.000000   
1/23/20  284.0  2.306338e+00  2.649984e+01    0.0000     0.000000   
1/24/20  284.0  3.313380e+00  3.299262e+01    0.0000     0.000000   
...        ...           ...           ...       ...          ...   
2/15/22  284.0  1.462300e+06  5.999364e+06    0.0000  7852.500000   
2/16/22  284.0  1.470893e+06  6.017744e+06    0.0000  7902.500000   
2/17/22  284.0  1.478046e+06  6.031356e+06    0.0000  7963.000000   
2/18/22  284.0  1.484923e+06  6.046653e+06    0.0000  7986.750000   
2/19/22  284.0  1.490334e+06  6.055395e+06    0.0000  7988.750000   

                  50%            75%           max  
Lat         21.607878      40.950592  7.170690e+01  
Long        20.921188      84.992575  1.780650e+02  
1/22/20      0.000000       0.000000  4.440000e+02  
1/23/20      0.000000       0.000000  4.440000e+02  
1/24/20      0.000000       0.000000  5.490000e+02  
...               ...            ...           ...  
2/15/22  63176.500000  734449.750000  7.803619e+07  
2/16/22  63193.500000  736980.250000  7.817436e+07  
2/17/22  63193.500000  738947.500000  7.827570e+07  
2/18/22  63234.000000  740284.250000  7.842352e+07  
2/19/22  63528.500000  741032.000000  7.845708e+07  

[762 rows x 8 columns]

In [72]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/10/22  2/11/22  \
0          0        0        0        0        0  ...   169448   169940   
1          0        0        0        0        0  ...   267020   267020   
2          0        0        0        0        0  ...   260191   260723   
3          0        0        0        0        0  ...    37074    37140   
4          0        0        0        0        0  ...    98474    98501   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      465      490   
281        0        0        0        0        0  ...    11604    11659   
282        0        0        0        0        0  ...   308556   309013   
283        0        0        0        0        0  ...   231040   231214   

     2/12/22  2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  
0     170152   170604   171246   171422   171519   171673   171857   171931  
1     267551   268008   268304   268491   268940   269301   269601   269904  
2     261226   261752   262165   262570   262994   263369   263685   263936  
3      37140    37140    37277    37361    37452    37522    37589    37589  
4      98514    98514    98514    98555    98568    98585    98605    98617  
..       ...      ...      ...      ...      ...      ...      ...      ...  
279   600128   600128   613846   618527   622175   625028   627956   627956  
280      498      501      501      505      507      507      508      508  
281    11664    11671    11679    11699    11707    11718    11731    11731  
282   309293   309502   309579   309870   310155   310474   310764   311016  
283   231299   231381   231603   231603   232213   232598   233030   233224  

[284 rows x 764 columns]

- Berapa banyak data yang kasus kumulatif pada tanggal 20 Januari untuk negara-negara dengan latitude di bawah 0?

In [73]:
data[data['Lat'] < 0][['1/20/22']]

1/20/22
4       95220
5          11
7     7576335
9       28472
10     919876
11       8219
12     298367
13      89043
14      23515
15     717692
16       1338
28     795087
30     243946
31   23595178
36      36868
58    1916522
94       7784
95      23244
96      84198
110    658045
116     68003
118     60509
121     47275
124     35189
125     14854
126    133617
130       454
132     45663
149   4277644
159    319379
160        37
168     31899
175     55827
176     83403
183     70094
191    221432
192    154897
199         2
200     15401
210     36446
211    516555
212   2780049
219    127227
223        13
229     34367
234        81
236   3572860
248     32393
250     19866
252         1
265        85
271         4
274    556163
276         7
282    299971
283    227552

In [74]:
data[data['Lat'] < 0][['1/20/22']].count()

1/20/22    56
dtype: int64

Ada 56 data negara-negara dengan latitude dibawah 0 untuk kasus kumulatif per tanggal 20 Januari 2022

# Answer
Menampilkan 5 data teratas untuk melihat gambaran dataset

In [75]:
data.head()

Province/State Country/Region       Lat       Long  1/22/20  1/23/20  \
0            NaN    Afghanistan  33.93911  67.709953        0        0   
1            NaN        Albania  41.15330  20.168300        0        0   
2            NaN        Algeria  28.03390   1.659600        0        0   
3            NaN        Andorra  42.50630   1.521800        0        0   
4            NaN         Angola -11.20270  17.873900        0        0   

   1/24/20  1/25/20  1/26/20  1/27/20  ...  2/10/22  2/11/22  2/12/22  \
0        0        0        0        0  ...   169448   169940   170152   
1        0        0        0        0  ...   267020   267020   267551   
2        0        0        0        0  ...   260191   260723   261226   
3        0        0        0        0  ...    37074    37140    37140   
4        0        0        0        0  ...    98474    98501    98514   

   2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  
0   170604   171246   171422   171519   171673   171857   171931  
1   268008   268304   268491   268940   269301   269601   269904  
2   261752   262165   262570   262994   263369   263685   263936  
3    37140    37277    37361    37452    37522    37589    37589  
4    98514    98514    98555    98568    98585    98605    98617  

[5 rows x 764 columns]

In [76]:
data_baru = data[data['Lat'] < 0][['1/20/22']]

In [77]:
data_baru

1/20/22
4       95220
5          11
7     7576335
9       28472
10     919876
11       8219
12     298367
13      89043
14      23515
15     717692
16       1338
28     795087
30     243946
31   23595178
36      36868
58    1916522
94       7784
95      23244
96      84198
110    658045
116     68003
118     60509
121     47275
124     35189
125     14854
126    133617
130       454
132     45663
149   4277644
159    319379
160        37
168     31899
175     55827
176     83403
183     70094
191    221432
192    154897
199         2
200     15401
210     36446
211    516555
212   2780049
219    127227
223        13
229     34367
234        81
236   3572860
248     32393
250     19866
252         1
265        85
271         4
274    556163
276         7
282    299971
283    227552

In [78]:
data_baru['1/20/22'].max()

23595178

Data kasus tertinggi pada tanggal 20 Januari 2022 untuk negara dengan dibawah latitude 0 adalah sebesar 23595178

In [79]:
data[data['Lat'] < 0][['1/20/22', "Country/Region"]].max()

1/20/22           23595178
Country/Region    Zimbabwe
dtype: object

Dari data diatas, negara Zimbabwe merupakan negara dengan kasus kumulatif paling tinggi pada tanggal 20 Januari 2022 untuk negara negara dengan latitude dibawah 0.

In [80]:
data.loc[data['Country/Region']=='Indonesia',:]['1/20/22']

149    4277644
Name: 1/20/22, dtype: int64

Data Indonesia pada tanggal 20 januari 2022 untuk kasus kumulatif adalah sebesar 4277644

In [81]:
data.head()

Province/State Country/Region       Lat       Long  1/22/20  1/23/20  \
0            NaN    Afghanistan  33.93911  67.709953        0        0   
1            NaN        Albania  41.15330  20.168300        0        0   
2            NaN        Algeria  28.03390   1.659600        0        0   
3            NaN        Andorra  42.50630   1.521800        0        0   
4            NaN         Angola -11.20270  17.873900        0        0   

   1/24/20  1/25/20  1/26/20  1/27/20  ...  2/10/22  2/11/22  2/12/22  \
0        0        0        0        0  ...   169448   169940   170152   
1        0        0        0        0  ...   267020   267020   267551   
2        0        0        0        0  ...   260191   260723   261226   
3        0        0        0        0  ...    37074    37140    37140   
4        0        0        0        0  ...    98474    98501    98514   

   2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  
0   170604   171246   171422   171519   171673   171857   171931  
1   268008   268304   268491   268940   269301   269601   269904  
2   261752   262165   262570   262994   263369   263685   263936  
3    37140    37277    37361    37452    37522    37589    37589  
4    98514    98514    98555    98568    98585    98605    98617  

[5 rows x 764 columns]

In [86]:
def Kasus_harian():
    "Menghitung kasus harian covid-19"
    Kasus = hari_ini - hari_kemarin
    return Kasus

In [82]:
data['Kasus harian'] = ''

In [83]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/11/22  2/12/22  \
0          0        0        0        0        0  ...   169940   170152   
1          0        0        0        0        0  ...   267020   267551   
2          0        0        0        0        0  ...   260723   261226   
3          0        0        0        0        0  ...    37140    37140   
4          0        0        0        0        0  ...    98501    98514   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      490      498   
281        0        0        0        0        0  ...    11659    11664   
282        0        0        0        0        0  ...   309013   309293   
283        0        0        0        0        0  ...   231214   231299   

     2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  \
0     170604   171246   171422   171519   171673   171857   171931   
1     268008   268304   268491   268940   269301   269601   269904   
2     261752   262165   262570   262994   263369   263685   263936   
3      37140    37277    37361    37452    37522    37589    37589   
4      98514    98514    98555    98568    98585    98605    98617   
..       ...      ...      ...      ...      ...      ...      ...   
279   600128   613846   618527   622175   625028   627956   627956   
280      501      501      505      507      507      508      508   
281    11671    11679    11699    11707    11718    11731    11731   
282   309502   309579   309870   310155   310474   310764   311016   
283   231381   231603   231603   232213   232598   233030   233224   

     Kasus harian  
0                  
1                  
2                  
3                  
4                  
..            ...  
279                
280                
281                
282                
283                

[284 rows x 765 columns]

In [92]:
data['Kasus harian'] = data['2/19/22'] - data['2/18/22']

In [93]:
data['Kasus harian']

0       74
1      303
2      251
3        0
4       12
      ... 
279      0
280      0
281      0
282    252
283    194
Name: Kasus harian, Length: 284, dtype: int64

Data kasus harian pertanggal 20 Februari 2022

In [90]:
data

Province/State        Country/Region        Lat        Long  1/22/20  \
0              NaN           Afghanistan  33.939110   67.709953        0   
1              NaN               Albania  41.153300   20.168300        0   
2              NaN               Algeria  28.033900    1.659600        0   
3              NaN               Andorra  42.506300    1.521800        0   
4              NaN                Angola -11.202700   17.873900        0   
..             ...                   ...        ...         ...      ...   
279            NaN    West Bank and Gaza  31.952200   35.233200        0   
280            NaN  Winter Olympics 2022  39.904200  116.407400        0   
281            NaN                 Yemen  15.552727   48.516388        0   
282            NaN                Zambia -13.133897   27.849332        0   
283            NaN              Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  2/11/22  2/12/22  \
0          0        0        0        0        0  ...   169940   170152   
1          0        0        0        0        0  ...   267020   267551   
2          0        0        0        0        0  ...   260723   261226   
3          0        0        0        0        0  ...    37140    37140   
4          0        0        0        0        0  ...    98501    98514   
..       ...      ...      ...      ...      ...  ...      ...      ...   
279        0        0        0        0        0  ...   600128   600128   
280        0        0        0        0        0  ...      490      498   
281        0        0        0        0        0  ...    11659    11664   
282        0        0        0        0        0  ...   309013   309293   
283        0        0        0        0        0  ...   231214   231299   

     2/13/22  2/14/22  2/15/22  2/16/22  2/17/22  2/18/22  2/19/22  \
0     170604   171246   171422   171519   171673   171857   171931   
1     268008   268304   268491   268940   269301   269601   269904   
2     261752   262165   262570   262994   263369   263685   263936   
3      37140    37277    37361    37452    37522    37589    37589   
4      98514    98514    98555    98568    98585    98605    98617   
..       ...      ...      ...      ...      ...      ...      ...   
279   600128   613846   618527   622175   625028   627956   627956   
280      501      501      505      507      507      508      508   
281    11671    11679    11699    11707    11718    11731    11731   
282   309502   309579   309870   310155   310474   310764   311016   
283   231381   231603   231603   232213   232598   233030   233224   

     Kasus harian  
0              74  
1             303  
2             251  
3               0  
4              12  
..            ...  
279             0  
280             0  
281             0  
282           252  
283           194  

[284 rows x 765 columns]

In [87]:
data.loc[data['Country/Region']=='Indonesia',:].max()

Province/State         None
Country/Region    Indonesia
Lat                 -0.7893
Long               113.9213
1/22/20                   0
                    ...    
2/16/22             4966046
2/17/22             5030002
2/18/22             5089637
2/19/22             5149021
Kasus harian               
Length: 765, dtype: object

In [95]:
df = pd.DataFrame(data)

In [99]:
df.diff(axis=1)

TypeError: unsupported operand type(s) for -: 'float' and 'str'